cirq/experiments/generate_google_v2_random_circuit.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from typing import Callable, Iterable, TypeVar
+
+import cirq
+from cirq.circuits import InsertStrategy
+from cirq.devices.grid_qubit import GridQubit
+from cirq.ops import common_gates
+
+
+def generate_random_v2_circuit(qubits: Iterable[GridQubit],
+                               cz_depth: int,
+                               seed: int) -> cirq.Circuit:
+
+
+    ND_GATES = [common_gates.X**(1/2), common_gates.Y**(1/2)]
+    rand_gen = random.Random(seed).random
+
+    circuit = cirq.Circuit()
+
+    # Add an initial moment of Hadamards
+    circuit.append(common_gates.H(qubit) for qubit in qubits)
+
+    layer_index = 0
+    if cz_depth:
+        layer_index = _add_cz_layer(layer_index, circuit)
+        # In the first moment, add T gates when possible
+        for qubit in qubits:
+            if not circuit.operation_at(qubit, 1):
+                circuit.append(common_gates.T(qubit),
+                               strategy=InsertStrategy.EARLIEST)
+
+    for moment_index in range(2, cz_depth+1):
+        layer_index = _add_cz_layer(layer_index, circuit)
+        # Add single qubit gates in the same moment
+        for qubit in qubits:
+            if not circuit.operation_at(qubit, moment_index):
+                last_gate = circuit.operation_at(qubit, moment_index-1)
+                if last_gate:
+                    # Add a random non diagonal gate after a CZ
+                    if last_gate.gate == common_gates.CZ:
+                        circuit.append(_choice(rand_gen, ND_GATES).on(qubit),
+                                       strategy=InsertStrategy.EARLIEST)
+                    # Add a T gate after a non diagonal gate
+                    elif not last_gate.gate == common_gates.T:
+                        circuit.append(common_gates.T(qubit),
+                                       strategy=InsertStrategy.EARLIEST)
+
+    # Add a final moment of Hadamards
+    circuit.append(common_gates.H(qubit) for qubit in qubits)
+
+    return circuit
+
+
+def generate_random_v2_circuit_grid(n_rows: int, n_cols: int,
```

Comment:
`v2` isn't going to mean anything to people looking over the documentation. Rename this method to something like `generate_supremacy_circuit_1807_10749`.

Reply:
v2 of course means v2, using the alternative you suggest will require a new paper if we want to update to v3

Reply:
Sure, `v2` means `v2`, but in what context? What is is the second version of? Users aren't going to see this code already-knowing all of the surrounding context, they're going to see it when browsing through the various methods exported by cirq. And in that context `v2` makes you think that it refers to version 2 of *cirq* or of *this method within cirq*, instead of some external paper they've not heard of up to this point.

How about `generate_google_v2_supremacy_circuit` or `generate_supremacy_circuit_1807_10749_v2`?

`generate_random_v2_circuit_grid` just isn't a good enough name.

Reply:
I am renaming this